ee/clickhouse/client.py

```diff
@@ -72,6 +75,29 @@ def async_execute(query, args=None):
     )
 
     def sync_execute(query, args=None):
-        with ch_sync_pool.get_client() as client:
-            result = client.execute(query, args)
+        start_time = time()
+        try:
+            with ch_sync_pool.get_client() as client:
+                result = client.execute(query, args)
+        finally:
+            execution_time = time() - start_time
+            if settings.SHELL_PLUS_PRINT_SQL:
+                print(format_sql(query, args))
+                print("Execution time: %.6fs" % (execution_time,))
         return result
+
+
+def format_sql(parameterized_sql, params):
+    params = {key: repr(value) for key, value in (params or {}).items()}
```

Comment:
You can just do `client.substitute_params(query, params)`

Reply:
Nice, thanks!